sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1408,6 +1408,12 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         DayOfMonth(expression(ctx.source))
       case "DAYOFWEEK" =>
         DayOfWeek(expression(ctx.source))
+      case "DOW" =>
+        Subtract(DayOfWeek(expression(ctx.source)), Literal(1))
```

Comment:
It's very confusing that `DOW` returns different results from `DAYOFWEEK`. `DOW` should just be an abbreviation of `DAYOFWEEK`. See also https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts

Do we have a clear document about what these names mean? cc @HyukjinKwon @maropu @wangyum

Reply:
Here we just follow PostgreSQL definition of `DOW`, see:
```
dow
The day of the week as Sunday (0) to Saturday (6)

SELECT EXTRACT(DOW FROM TIMESTAMP '2001-02-16 20:38:40');
Result: 5
```
Do you propose to break compatibility with PostgreSQL?

Reply:
We never promise pgsql compatibility and internal consistency is much more important for Spark.

For this case, I think the behavior of `DOW` from pgsql is correct and commonly used by other databases. We should fix our `DAYOFWEEK` instead.

Reply:
> Do we have a clear document about what these names mean? cc @HyukjinKwon @maropu @wangyum

Currently, we just list up these names in `ExpressionDescription` of `DatePart`: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2124
If [we add `ExpressionDescription` for `EXTRACT`](https://github.com/apache/spark/pull/21479#discussion_r409933251), the list will appear for `EXTRACT`, too, in [the Built-in Functions document ](https://spark.apache.org/docs/latest/api/sql/index.html) for Spark 3.0. If we need to describe more about these parameters, I think we need to update the `argument` section there.

Reply:
BTW, @cloud-fan . `DAYOFWEEK` is added two years ago. (https://github.com/apache/spark/pull/21479)
> We should fix our DAYOFWEEK instead.

cc @marmbrus and @gatorsmile since the request seems to ask a behavior change.